### ISSUE TITLE
Replace xpm icon by png

### ISSUE THREAD

**Issue by AntoineTurmel** [nightingale-media-player/nightingale-hacking#258]
Mozilla based products ships with png icons, often mozicon128
firefox/browser/icons/mozicon128.png
thunderbird/chrome/icons/default/default16.png (16,22,24,32,48,256)
instantbird/icons/mozicon128.png
instantbird/chrome/icons/default/*.png

We only distribute a xpm which looks buggy on gnome-panel:
![ngale_icon_xpm](https://f.cloud.github.com/assets/119288/2062324/4318bec0-8c86-11e3-9d5b-9557678ea682.png)
(between terminal icon and spotify icon)

also the icon don't display on taskbar:
![taskbar](https://f.cloud.github.com/assets/119288/2062376/2d7eb148-8c89-11e3-9a44-ddf195256552.png)
##

**Comment by darealshinji** [comment]
I this fixed now that all the icons with different sizes are installed by default on Linux?
https://github.com/nightingale-media-player/nightingale-hacking/blob/sb-trunk-oldxul/Makefile#L235

**Comment by freaktechnik** [comment]
I think the xpm is still built into the nightingale executable, and nobody's figured out how to change that yet. (It's in the Makefile or jar.mn in app/branding)

**Comment by PanderMusubi** [comment]
On Ubuntu 15.04 GNOME with Nightingale 1.13a this seems to work OK. Nightingale ships:

```
$ dpkg -L nightingale|grep png
/usr/share/icons/hicolor/96x96/apps/nightingale.png
/usr/share/icons/hicolor/16x16/apps/nightingale.png
/usr/share/icons/hicolor/32x32/apps/nightingale.png
/usr/share/icons/hicolor/48x48/apps/nightingale.png
/usr/share/icons/hicolor/64x64/apps/nightingale.png
/usr/share/icons/hicolor/24x24/apps/nightingale.png
/usr/share/icons/hicolor/512x512/apps/nightingale.png
/usr/share/icons/hicolor/256x256/apps/nightingale.png
/usr/share/icons/hicolor/128x128/apps/nightingale.png
/usr/lib/nightingale/xulrunner/chrome/icons/default/default48.png
/usr/lib/nightingale/xulrunner/chrome/icons/default/default32.png
/usr/lib/nightingale/xulrunner/chrome/icons/default/default16.png
/usr/lib/nightingale/xulrunner/res/html/folder.png
/usr/lib/nightingale/xulrunner/res/broken-image.png
/usr/lib/nightingale/xulrunner/res/loading-image.png
/usr/lib/nightingale/xulrunner/icons/document.png
/usr/lib/nightingale/icons/updater.png
$ dpkg -L nightingale|grep xpm
/usr/lib/nightingale/chrome/icons/default/default.xpm
```

@AntoineTurmel, is this still a problem for you in Unity? See also extensions/unity-integration/README

Nevertheless, figuring out what is build into the executable should be traceable and fixed.

@freaktechnik, perhaps this might help locating it:

```
nightingale-hacking-sb-trunk-oldxul$ grep -riI xpm *
app/branding/Makefile.in:   ICON_EXTENSION = .xpm
app/branding/nightingale.xpm:/* XPM */
app/branding/nightingale.xpm:static char * nightingale_xpm[] = {
...
dependencies/vendor/mozbrowser/installer/package-manifest.in:@BINPATH@/icons/*.xpm
...
update/manifest.linux-i686:./chrome/icons/default/default.xpm
...
```

**Comment by freaktechnik** [comment]
@PanderMusubi: I don't think you understand the issue here. Executables can provide an icon that should be used instead of the default executable icon at runtime. Most desktop environements (like unity) don't care about this icon, if there's a desktop file for the executable specifying another icon. What needs to be done here is use the png in the nightingale executable instead of the xpm.

**Comment by PanderMusubi** [comment]
Yes, that is what I gathered. Tried to help out to figure out where that is set in the building process.
